pkg/operator/controller.go

```diff
@@ -133,6 +174,17 @@ func (c *Controller) sync() error {
 		return fmt.Errorf("unable to get cluster proxy configuration: %v", err)
 	}
 
+	// calculate current ca bundle hash.
+	if c.caBundleHash == "" {
+		klog.Info("calculating local ca bundle hash")
+		bundleHash, err := c.localCABundleHash()
```

Comment:
We don't want to compute this during `sync` (which is called every time the operator gets an event from the cluster) - the caBundleHash should be set once when the controller is initialized. The empty string is an acceptable value if the trust bundle hasn't been injected yet.